corehq/apps/accounting/tests/test_autopay_with_api.py

```diff
@@ -118,22 +119,37 @@ def test_get_autopayable_invoices_returns_nothing(self):
     def test_pay_autopayable_invoices(self):
         original_outbox_length = len(mail.outbox)
 
-        autopayable_invoice = Invoice.objects.filter(subscription=self.subscription)
-        date_due = autopayable_invoice.first().date_due
+        autopayable_invoice = Invoice.objects.filter(subscription=self.subscription).first()
+        date_due = autopayable_invoice.date_due
+
+        # Keys for idempotent requests can only be used with the same parameters they were first used with.
+        # So introduce randomness in idempotency key to avoid clashes
```

Comment:
Because idempotency key is [consisted of invoice number and the total amount we're going to charge.](https://github.com/dimagi/commcare-hq/blob/6e7975aec6094e6c9593415ddf3c7b67e977de31/corehq/apps/accounting/payment_handlers.py#L404C23-L404C23)